src/views/virtualmachines/details/tabs/configuration/network/hooks/useInterfacesAndNetworks.ts

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from 'react';
+
+import {
+  V1Interface,
+  V1Network,
+  V1VirtualMachine,
+  V1VirtualMachineInstance,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getInterfaces, getNetworks } from '@kubevirt-utils/resources/vm/utils/selectors';
+import { getVMIInterfaces, getVMINetworks } from '@kubevirt-utils/resources/vmi/utils/selectors';
+import { removeDuplicatesByName } from '@kubevirt-utils/utils/utils';
+
+type UseInterfacesAndNetworks = (
+  vm: V1VirtualMachine,
+  vmi: V1VirtualMachineInstance,
+) => {
+  interfaces: V1Interface[];
+  networks: V1Network[];
+};
+
+const useInterfacesAndNetworks: UseInterfacesAndNetworks = (vm, vmi) => {
+  return useMemo(() => {
+    const vmNetworks = getNetworks(vm) || [];
+    const vmInterfaces = getInterfaces(vm) || [];
+
+    const vmiInterfaces = getVMIInterfaces(vmi) || [];
+    const vmiNetworks = getVMINetworks(vmi) || [];
+
+    const networks = removeDuplicatesByName([...vmNetworks, ...vmiNetworks]);
+    const interfaces = removeDuplicatesByName([...vmInterfaces, ...vmiInterfaces]);
+
+    return {
+      interfaces,
+      networks,
+    };
+  }, [vm, vmi]);
+};
```

Comment:
useMemo is good when there is a heavy calculation or too much rendering, which could help prevent it. I don't think this is the case. Please feel free to check. Just make this a util function instead of the hook. I guess it will work the same without all the boilerplate of a hook.

Reply:
Gotcha. Updated. I also changed the name of a function in the file that I added the utility function to because it was a bad name. That caused an update in three hook files that are all exactly the same. Was having the same file copied three times intentional?